Issue/175 expand api to pull change codes update display

SavvyDolphin77 opened the pull request:
# Proposed Changes

* API now extracts transaction codes
* Recent ownerships changed the name on page to be more appropriate and clear
* Updated display to display transaction code rather then the form type/number
* Added information below title to explain what the section was about and more about the codes
* Added fucntion to display if the code was a code meaning person gained, loss or nothing

Fixes #175  .

SavvyDolphin77 committed:
Adds API to Extract Transaction Codes
Extracts transaction codes from the XLM in the API call and returns it.

SavvyDolphin77 committed:
Adds Information & Replaces Form with Code
Adds information under the recent ownership section to explain the codes and information there.
Changes name of recent ownerships to fit what the section is about more clearly.
Replaces the form column in the code with the transaction code.
Adds function using our analysis to add if tranaction code is gaining or losing control

JimtheBurger commented on adversarial_apps/src/components/RecentOwnership.tsx:
```diff
@@ -126,7 +151,7 @@ export const RecentOwnership = (props: Props) => {
                                     </td>
                                     <td className="px-6 py-4 whitespace-nowrap text-sm">
-                                        {filing.form}
+                                        {filing.transaction_code} {getTransactionSymbol(filing.transaction_code)}
```
Pretty sure for the interface on line 10-16 you want it to be:
```
interface FilingReport {
    transaction_code: string;
    accessionNumber: string;
    fillingDate: string;
    form: string;
    issuer: string;
    reporter: string;
}
```

This way the interface is now expecting the `transaction_code` along with the other responses.

JimtheBurger commented on adversarial_apps/src/components/RecentOwnership.tsx:
```diff
@@ -90,7 +102,20 @@ export const RecentOwnership = (props: Props) => {
+                The letter relates to the exact type of transaction that has occurred. For covnince we summed up the transaction codes and provided the key:
+            </p>
+            <ul className="list-disc ml-5">
+                <li>If the transaction code letter has a + that means the Receiver has obtained more control over the company.</li>
+                <li>If the transaction code letter has a - that means the Receiver has lost or given up control over the company.</li>
+                <li>If the transaction code letter has a ~ that means the Receiver has not changed their ownership of the company.</li>
```
nitpick: would like if the +,-,~ were bolded or in quotes like this to help them stand out:
```suggestion
                <li>If the transaction code letter has a "+" that means the Receiver has obtained more control over the company.</li>
                <li>If the transaction code letter has a "-" that means the Receiver has lost or given up control over the company.</li>
                <li>If the transaction code letter has a "~" that means the Receiver has not changed their ownership of the company.</li>
```

JimtheBurger commented on adversarial_apps/src/components/RecentOwnership.tsx:
```diff
@@ -106,13 +131,13 @@ export const RecentOwnership = (props: Props) => {
                                 </th>
                                 <th className="px-6 py-3 text-left text-xs font-medium uppercase tracking-wider">
-                                    Form
+                                    Transaction Type
```
I LOVE this!

I think there is the potential to have this hoverable like our `RiskScoreTooltip.tsx`. Specifically to explain what S but **totally not necessary for this PR if you don't want to imo**.

If it were to be added I think it would be to explain what the codes mean:
```
const gainingCodes = ["P", "A", "M", "C", "O", "X", "L"];
const losingCodes = ["S", "D", "F", "H", "G", "U"];
const neutralCodes = ["V", "I", "E", "W", "Z", "J", "K"];
```

SavvyDolphin77 replied:
Moving this to a new PR/issue

JimtheBurger reviewed (changes requested):
Really Good! I love this and this will look outstanding in a presentation format. Just a few request changes requested, but otherwise good.

SavvyDolphin77 committed:
Fixed Changes Requested: Interface Transaction
Updated transaction code information to "" the types of symbols we use.
Added the transaction_code to the interface for ensuring correct type

SavvyDolphin77 committed:
Fix eslint Errors

SavvyDolphin77 committed:
Fix eslint Errors Again
First fix didn't work, here is second fix.
Fixes qoutes being escapable in the code.
